integrations/spiral/src/Bootloader/SearchBootloader.php

```diff
@@ -82,144 +95,82 @@ public function boot(ConsoleBootloader $console, Container $container): void
+            $adapterServiceId = 'schranz_search.adapter.'.$name;
+            $engineServiceId = 'schranz_search.engine.'.$name;
+            $schemaLoaderServiceId = 'schranz_search.schema_loader.'.$name;
+            $schemaId = 'schranz_search.schema.'.$name;
 
             /** @var string $adapterDsn */
             $adapterDsn = $engineConfig['adapter'];
             $dirs = $engineSchemaDirs[$name] ?? [];
 
-            $container->bindSingleton($adapterServiceId, function (Container $container) use ($adapterDsn) {
-                /** @var AdapterFactory $factory */
-                $factory = $container->get(AdapterFactory::class);
-
-                return $factory->createAdapter($adapterDsn);
-            });
+            $container->bindSingleton($adapterServiceId, $adapter = $adapterFactory->createAdapter($adapterDsn));
+            $container->bindSingleton($schemaLoaderServiceId, $loader = new PhpFileLoader($dirs, $config->getPrefix()));
+            $container->bindSingleton($schemaId, $schema = $loader->load());
+            $container->bindSingleton($engineServiceId, $engine = new Engine($adapter, $schema));
 
-            $container->bindSingleton($schemaLoaderServiceId, fn (Container $container) => new PhpFileLoader($dirs, $config->getPrefix()));
-
-            $container->bindSingleton($schemaId, function (Container $container) use ($schemaLoaderServiceId) {
-                /** @var LoaderInterface $loader */
-                $loader = $container->get($schemaLoaderServiceId);
-
-                return $loader->load();
-            });
-
-            $engineServices[$name] = $engineServiceId;
-            $container->bindSingleton($engineServiceId, function (Container $container) use ($adapterServiceId, $schemaId) {
-                /** @var AdapterInterface $adapter */
-                $adapter = $container->get($adapterServiceId);
-                /** @var Schema $schema */
-                $schema = $container->get($schemaId);
-
-                return new Engine($adapter, $schema);
-            });
-
-            if ('default' === $name || (!isset($engines['default']) && !$container->has(Engine::class))) {
-                $container->bind(Engine::class, $engineServiceId);
-            }
+            $engineServices[$name] = $engine;
         }
 
-        $container->bindSingleton(EngineRegistry::class, function (Container $container) use ($engineServices) {
-            $engines = [];
-
-            foreach ($engineServices as $name => $engineServiceId) {
-                /** @var Engine $engine */
-                $engine = $container->get($engineServiceId);
-
-                $engines[$name] = $engine;
-            }
-
-            return new EngineRegistry($engines);
-        });
+        return new EngineRegistry($engineServices);
     }
 
-    private function createAdapterFactories(Container $container): void
+    private function defaultEngine(EngineRegistry $registry): Engine
```

Comment:
Not necessary as this should as written be handled differently if there is no `default` define still a default engine is created, all integration should work the same.